services/frontend-service/src/ui/utils/store.tsx

```diff
@@ -175,9 +174,34 @@ export const addAction = (action: BatchAction): void => {
     UpdateSidebar.set({ shown: true });
 };
 
-export const updateReleaseDialog = (app: string, version: number): void => {
-    UpdateReleaseDialog.set({ app: app, version: version });
+export const useOpenReleaseDialog = (app: string, version: number): (() => void) => {
+    const [params, setParams] = useSearchParams();
+    return useCallback(() => {
+        params.set('dialog-app', app);
+        params.set('dialog-version', version.toString());
+        setParams(params);
+    }, [app, params, setParams, version]);
+};
+
+export const useCloseReleaseDialog = (): (() => void) => {
+    const [params, setParams] = useSearchParams();
+    return useCallback(() => {
+        params.delete('dialog-app');
+        params.delete('dialog-version');
+        setParams(params);
+    }, [params, setParams]);
+};
+
+export const useValidReleaseDialogParams = (): [string, number] => {
```

Comment:
please return a struct instead of an array, to make it more readable.

Reply:
done :+1: